users/views.py

```diff
@@ -60,6 +60,25 @@ def update(self, request, *args, **kwargs):
             return update_result
 
 
+class UserInfoViewSet(CurrentUserRetrieveUpdateViewSet):
+    """
+    User info viewset for the open edx OAuth, extends CurrentUserRetrieveUpdateViewSet
+    """
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Retrieve the current user's info only if they have an edx_username, otherwise return 409
+
+        This is to prevent issues with Open edX OAuth client that expect an edx_username to be present
+        """
+        user = self.get_object()
+        if user.is_authenticated and user.edx_username:
+            return super().retrieve(request, *args, **kwargs)
+        return Response(
+            {"get": "User has no edx_username."}, status=status.HTTP_409_CONFLICT
```

Comment:
Is the key of `"get"` significant to openedx? Would be helpful to note that otherwise I'd suggest `"error"` is more apt here.